rskj-core/src/main/java/co/rsk/peg/feeperkb/FeePerKbSupport.java

```diff
@@ -0,0 +1,105 @@
+package co.rsk.peg.feeperkb;
+
+import co.rsk.peg.abi.*;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.peg.AddressBasedAuthorizer;
+import co.rsk.peg.BridgeSerializationUtils;
+import co.rsk.peg.feeperkb.constants.*;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+public class FeePerKbSupport {
+    private static final Integer FEE_PER_KB_GENERIC_ERROR_CODE = FeePerKbResponseCodes.getGenericErrorCode();
+    private static final Integer NEGATIVE_FEE_PER_KB_ERROR_CODE = FeePerKbResponseCodes.getNegativeFeeErrorCode();
+    private static final Integer EXCESSIVE_FEE_PER_KB_ERROR_CODE = FeePerKbResponseCodes.getNegativeFeeErrorCode();
+
+    private final FeePerKbStorageProvider provider;
+    private final FeePerKbConstants feePerKbConstants;
+    private static final Logger logger = LoggerFactory.getLogger("FeePerKbSupport");
+
+    public FeePerKbSupport(FeePerKbConstants feePerKbConstants, FeePerKbStorageProvider provider) {
+        this.provider = provider;
+        this.feePerKbConstants = feePerKbConstants;
+    }
+
+    /**
+     * @return Current fee per kb in BTC.
+     */
+    public Coin getFeePerKb() {
+        Coin currentFeePerKb = provider.getFeePerKb();
+
+        if (currentFeePerKb == null) {
+            currentFeePerKb = feePerKbConstants.getGenesisFeePerKb();
+        }
+
+        return currentFeePerKb;
+    }
+
+    /**
+     * Votes for a fee per kb value.
+     *
+     * @return 1 upon successful vote, -1 when the vote was unsuccessful,
+     * FEE_PER_KB_GENERIC_ERROR_CODE when there was an un expected error.
+     */
+    public Integer voteFeePerKbChange(Transaction tx, Coin feePerKb, SignatureCache signatureCache) {
+
+        AddressBasedAuthorizer authorizer = feePerKbConstants.getFeePerKbChangeAuthorizer();
+        Coin maxFeePerKb = feePerKbConstants.getMaxFeePerKb();
+
+        if (!authorizer.isAuthorized(tx, signatureCache)) {
+            return FEE_PER_KB_GENERIC_ERROR_CODE;
+        }
+
+        if(!feePerKb.isPositive()){
+            return NEGATIVE_FEE_PER_KB_ERROR_CODE;
+        }
+
+        if(feePerKb.isGreaterThan(maxFeePerKb)) {
+            return EXCESSIVE_FEE_PER_KB_ERROR_CODE;
+        }
```

Comment:
I would add logs here for each of these cases

Reply:
Also in the rest of the method. And update the existing logs, add the `[METHOD_NAME]` prefix